drivers/ethernet/phy/phy_mii.c

```diff
@@ -30,10 +32,12 @@ struct phy_mii_dev_data {
 	const struct device *dev;
 	phy_callback_t cb;
 	void *cb_data;
-	struct k_work_delayable monitor_work;
 	struct phy_link_state state;
 	struct k_sem sem;
+#if ONLY_FIXED_LINK
+	struct k_work_delayable monitor_work;
 	bool gigabit_supported;
+#endif
```

Comment:
I don't understand , isn't the use of this macro backwards in the entire file? You are including the code for monitoring and reading mdio if it *is* only a fixed link?

Reply:
Yeah it was the wrong way, I fixed it.